components/raftstore/src/store/worker/read.rs

```diff
@@ -763,20 +867,22 @@ where
                             }
                             None => monotonic_raw_now(),
                         };
-                        if !delegate.is_in_leader_lease(snapshot_ts) {
+
+                        let mut snap_cache_ctx =
+                            SnapCacheContext::new(&mut self.snap_cache, read_id);
+                        snap_cache_ctx.maybe_update_snapshot(delegate.get_tablet());
+
+                        let region = Arc::clone(&delegate.region);
+                        let response =
+                            delegate.execute(&req, &region, None, Some(&mut snap_cache_ctx));
```

Comment:
Should check lease before executing requests.

Reply:
Good catch.